lighthouse-core/audits/accessibility/axe-audit.js

```diff
@@ -39,8 +39,20 @@ class AxeAudit extends Audit {
       };
     }
 
-    const violations = artifacts.Accessibility.violations || [];
-    const rule = violations.find(result => result.id === this.meta.id);
+    // Detect errors reported within aXe 'incomplete' results
+    // aXe uses this result type to indicate errors, or rules which require manual investigation
+    // If aXe reports an error, then bubble it up to the caller
+    const incomplete = artifacts.Accessibility.incomplete || [];
+    const isIncomplete = incomplete.find(result => result.id === this.meta.id);
+    if (isIncomplete && isIncomplete.error) {
+      return Audit.generateErrorAuditResult(this, isIncomplete.error.message);
+    }
+
+    const failureCases = [
```

Comment:
after reading through the [axe docs](https://github.com/dequelabs/axe-core/blob/6b444546cff492a62a70a74a8fc3c62bd4729400/doc/API.md#results-object) on this property (a great link we should probably throw into a comment here 😃), I still think we should only merge the non-error `.incomplete` results in if it's an informative audit. It will still get the same treatment for the elements raised and "passing" becomes "notApplicable" for those audits instead, but it seems to be the equivalent semantic relationship we have to their "review items".

WDYT?

Reply:
That does makes sense, yep - I'll try including that here.  Out of interest - I did notice there's also a `manual` scoring mode - is `informative` definitely the preferable one here?

I'd also be curious if you have any suggestions on the variable naming of `failureCases`.  I'm not sure that's ideal and if you'd have any domain knowledge on improvements that could be handy.

Reply:
Getting closer to intended behaviour now I believe.  From local testing via DBW:

- [x] No form elements on the page?  Audit will fall under 'Not Applicable'
- [x] Form elements exist, with no automatically-determined label failures?  Audit displays as informative, and no score penalty is applied
- [x] Form elements exist, with multiple labels detected by axe?  Audit displays as informative, and lists the offending elements, and no score penalty is applied

The second scenario above still puzzles me a bit, but I *think* fits with the spirit of `incomplete` requiring manual review.

It could be collapsed into the third scenario by reintroducing the check on the `nodes.length` value.

Reply:
> I did notice there's also a manual scoring mode - is informative definitely the preferable one here?

Yes, `informative` is what we want here. `manual` is for audits that literally have no automatic insights whatsoever. It's essentially a text block of documentation to mark off a manual checklist.

> Form elements exist, with no automatically-determined label failures? Audit displays as informative, and no score penalty is applied

*I think* we should mark this one also as just `notApplicable` instead. To me that makes a little more sense than having an audit that shows up without pointing to any specific elements.

Other than that I agree I think we're good here. Great work!

Reply:
@patrickhulce Thanks!

For some reason I just can't seem to handle the second case the way I'd expect to.  As best as I can tell, `axe` must be returning the result inside the `pass` set if form labels exist, but definitively do *not* have multiple labels assigned to them.

In other words, either the test is `inapplicable` *or* `applicable && pass` *or* `applicable && incomplete`.

Adding the check for `incomplete && nodes` therefore has no effect.  I'll experiment with examining the contents of the `pass` set to try to verify this.

Reply:
My proposed behavior is something we would have to implement on our side, axe won't give it to us.

Basically `if (!failureCases.length && this.meta.scoreDisplayMode === 'informative') return someNotApplicableResult()`

Reply:
Almost there.  It is indeed the case that these cases fall into the `passes` set.  The `nodes` logic I just pushed isn't quite correct either though, which I just caught via DBW.  Taking a while to wrap my mind around the various nested boolean conditions/scenarios :)

Reply:
@patrickhulce Alrighty.  There was one more case to handle -- `axe` may return **both** passes and failures for an individual rule (for example, if some form elements on the page passed the single-label test, but some were considered fail/incomplete).

I've pushed changes which handle that, and also in a way which is backwards compatible across the `axe-core` version upgrade.  I imagine this changeset would be applied in tandem with the version bump, but the backwards compatibility aims to avoid any problems even if not (for example, if any selective merges/branching should occur).

Reply:
... all of which simplifies down to the logic you mentioned in your comment, which I missed yesterday :)